### PR TITLE
Insert prompts block count basis

### LINKED ISSUE #623: Block-count-based insertion
## As is

An inline prompt can be inserted based on percentage

## To be 

Inline prompts should be possible to insert at block-count positions, e.g. after 2nd block

### PR DESCRIPTION
**Pull request by kariae**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #623.

### How to test the changes in this Pull Request:

1. Load this PR.
2. Create a new inline prompt.
3. Set the trigger to `Blocks count` from the right sidebar, and set the number of blocks to be rendered before inserting the prompt.
4. Load a post and observe the prompt is inserted after the number of blocks set in 2.
5. Smock test some edge cases, like having the prompt after a heading (it shouldn't be inserted), the prompt as the first/last element on the content.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/editor/Sidebar.js [hunk 1]
```diff
@@ -156,13 +161,35 @@ const Sidebar = ( {
...
-					label={ __( 'Approximate Position (in percent)', 'newspack-popups' ) }
-					value={ trigger_scroll_progress }
-					onChange={ value => onMetaFieldChange( 'trigger_scroll_progress', value ) }
-					min={ 0 }
-					max={ 100 }
-				/>
+				<>
+					<SelectControl
+						label={ __( 'Trigger', 'newspack-popups' ) }
+						help={ __( 'The event to trigger the prompt.', 'newspack-popups' ) }
```
Comment (adekbadek) [inline src/editor/Sidebar.js:167]: This is not about an event, just an insertion point. 

```suggestion
						help={ __( 'The position at which to insert the prompt.', 'newspack-popups' ) }
```

File: src/editor/Sidebar.js [hunk 2]
```diff
@@ -156,13 +161,35 @@ const Sidebar = ( {
...
-				<RangeControl
-					label={ __( 'Approximate Position (in percent)', 'newspack-popups' ) }
-					value={ trigger_scroll_progress }
-					onChange={ value => onMetaFieldChange( 'trigger_scroll_progress', value ) }
-					min={ 0 }
-					max={ 100 }
-				/>
+				<>
+					<SelectControl
+						label={ __( 'Trigger', 'newspack-popups' ) }
```
Comment (adekbadek) [inline src/editor/Sidebar.js:166]: The overlay prompts are "triggered", since they are initially hidden. Inline prompts are just inserted into content.

```suggestion
						label={ __( 'Insertion position', 'newspack-popups' ) }
```

File: src/editor/Sidebar.js [hunk 3]
```diff
@@ -156,13 +161,35 @@ const Sidebar = ( {
...
+						help={ __( 'The event to trigger the prompt.', 'newspack-popups' ) }
+						value={ trigger_type }
+						options={ [
+							{ label: __( 'Scroll Progress', 'newspack-popups' ), value: 'scroll' },
```
Comment (adekbadek) [inline src/editor/Sidebar.js:170]: "Scroll progress" is more suitable in the case of overlay prompts, where the action of scrolling is triggering the prompt to be displayed. Here it's just about the insertion position.

```suggestion
							{ label: __( 'Percentage', 'newspack-popups' ), value: 'scroll' },
```

File: src/editor/utils.js [hunk 4]
```diff
@@ -235,14 +237,25 @@ export const getPlacementHelpMessage = (
...
 		case 'inline':
-			return sprintf(
-				// Translators: Trigger percentage.
-				__(
-					'The prompt will be automatically inserted about %s into article content.',
-					'newspack-popups'
-				),
-				triggerPercentage + '%'
-			);
+			return triggerCount
```
Comment (adekbadek) [inline src/editor/utils.js:240]: This is not an accurate condition. When switching between percentage-insertion and block-count-insertion, `triggerCount` will stay the same and so the help text will not be updated:

image

The condition should be based on the `trigger_type` meta.

File: includes/class-newspack-popups.php [hunk 5]
```diff
@@ -152,6 +153,18 @@ public static function register_meta() {
 			]
 		);
 
+		\register_meta(
+			'post',
+			'trigger_blocks_count',
```
Comment (adekbadek) [inline includes/class-newspack-popups.php:158]:
```suggestion
			// Not really a "trigger", since this meta applies only to inline prompts. Keeping the "trigger"-based naming for consistency.
			'trigger_blocks_count',
```

**Review by adekbadek — CHANGES_REQUESTED** [review]
The "trigger"-based naming in the code is misleading, but this is caused by `trigger_scroll_progress` meta being reused by overlay & inline prompts. Ideally, the name of the new meta should be something more like `insertion_blocks_count`, but on the other hand the `trigger_*` meta names do form a group here. Just pointing this out for a possible future refactoring.

**Review by adekbadek — APPROVED** [review]
